src/web-animations-next-animation.js

```diff
@@ -112,13 +150,75 @@
     _arrangeChildren: function(childAnimation, offset) {
       if (this.startTime === null) {
         childAnimation.currentTime = this.currentTime - offset / this.playbackRate;
-        childAnimation._startTime = null;
       } else if (childAnimation.startTime !== this.startTime + offset / this.playbackRate) {
         childAnimation.startTime = this.startTime + offset / this.playbackRate;
       }
     },
     get playState() {
-      return this._animation.playState;
+      return this._animation ? this._animation.playState : 'idle';
+    },
+    _resetFinishedPromise: function() {
+      if (scope.animationsWithPromises.indexOf(this) == -1)
+        scope.animationsWithPromises.push(this);
+      this._finishedPromise = new Promise(
+          function(resolve, reject) {
+            this._finishedPromiseState = 'pending';
+            this._resolveFinishedPromise = function() {
+              this._finishedPromiseState = 'resolved';
+              resolve(this);
+            };
+            this._rejectFinishedPromise = function() {
+              this._finishedPromiseState = 'rejected';
+              // FIXME: If reject is not specified via `then`, this throws the error inside the
```

Comment:
I don't understand your FIXME. If this is called but I've yet to specify a `.catch` handler (shorthand for the second param to `.then`), this throws on line 174? That doesn't sound right.